File: java/src/test/java/mx/iteso/GoogleDocument_Test.java
package mx.iteso;

import static org.junit.Assert.assertEquals;

import java.awt.Font;

import org.junit.Test;

/**
 * Google Document Test Class.
 */
public class GoogleDocument_Test {

    /**
     * Google document.
     */
    private GoogleDoc googleDoc = new GoogleDoc();

    /**
     * Style evaluator.
     */
    @Test
    public void testGetStyle() {
        Format style = googleDoc.getStyle();
        assertEquals("Roboto", style.getFont().getName());
    }

    /**
     * Background test.
     */
    @Test
    public void testgetBackground() {
        BackgrdImage bg = googleDoc.getBackground();
        assertEquals("image.png", bg.getImage().getUrl());
    }

    /**
     * Set sharing permission test.
     */
    @Test
    public void testsetSharingPermissions() {
        googleDoc.setLicensePermissions(0);
        assertEquals(0, googleDoc.getLicensePermissions());
    }

    /**
     * Font evaluator.
     */
    @Test
    public void testGetFont() {
        Font font = googleDoc.getFont();
        assertEquals("Roboto", font.getFontName());
    }


}

File: python/src/unittest/python/GoogleDocumentTest.py
from GoogleDocument import *
from BackImage import *
from Format import *
from Font import *
import unittest


class GoogleDocumentTest(unittest.TestCase):

    def testPermissions(self):
        self.google_doc.setPermissions(10)
        self.assertEquals(10, self.google_doc.getPermissions())


    def testFont(self):
        self.assertEquals(Font('Arial', 11).getFontText(),
                          self.google_doc.getFont().getFontText())

    def testStyle(self):
        self.assertEquals(Format().getFont().getFontText(),
                          self.google_doc.getStyle()
                          .getFont().getFontText())


    def setUp(self):
        self.google_doc = GoogleDocument()


    def testBackground(self):
        self.assertEquals(BackgroundImage().getImage().get_url(),
                          self.google_doc
                          .getBackground().getImage().get_url())

